server/in_memory_state.go

```diff
@@ -0,0 +1,367 @@
+package server
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/docker/docker/pkg/registrar"
+	"github.com/docker/docker/pkg/truncindex"
+	"github.com/kubernetes-incubator/cri-o/oci"
+)
+
+// TODO: make operations atomic to greatest extent possible
+
+// InMemoryState is an in-memory state store, suitable for use when no other
+// programs are expected to interact with the server
+type InMemoryState struct {
+	lock         sync.Mutex
+	sandboxes    map[string]*sandbox
+	containers   oci.ContainerStorer
```

Comment:
Using an abstract oci.ContainerStorer implementation here is pretty useless. The MemoryStore we use is just a single map, plus a mutex which we don't need (because we have our own lock), and we only use trivial methods of ContainerStorer. Just use a:
```
	containers map[string]*Container
```